provider/consul_catalog.go

```diff
@@ -92,9 +93,11 @@ func (provider *ConsulCatalog) watchServices(stopCh <-chan struct{}) <-chan map[
 			default:
 			}
 
-			data, meta, err := catalog.Services(opts)
+			// Listening to changes that leads to `passing` state or degrades from it.
+			// The call is used just as a trigger for further actions (intentionally there is no interest in the received data).
+			_, meta, err := health.State("passing", opts)
```

Comment:
@emilevauge
It is a valid point!
I can confirm that trigger wont work in this situation. Should rework it to listen both the `catalog.Services` and `health.State` at the same time